sympy/printing/str.py

```diff
@@ -54,7 +54,7 @@ def _print_Add(self, expr, order=None):
         l = []
         for term in terms:
             t = self._print(term)
-            if t.startswith('-'):
+            if t.startswith('-') and not term.is_Add:
```

Comment:
Any reason to use `is_Add` here and `isinstance(term, Add)` below line 62? I think should both be `is_Add`

Reply:
@mloubout I have made the changes you asked for, please let me know if that does the work.

Reply:
This change suggested by @smichr in https://github.com/sympy/sympy/issues/24022 appears to work fine, but more tests are needed to validate the fix.

Also, while we are at it, let's rename `PREC` to `prec` on line 53
```
PREC = precedence(expr)
```
and then on line 62:
```
if precedence(term) < PREC or term.is_Add:
```